src/main/java/org/codehaus/plexus/components/io/resources/PlexusIoFileResourceCollection.java

```diff
@@ -279,4 +282,26 @@ public void forEach( PlexusIoResourceConsumer resourceConsumer )
             return result.iterator();
         }
     }
+    
+    /**
+     * Add a suffix to fileName (eg : test.xml => testSuffix.xml)
+     */
+    private String addFileNameSuffix(String name)
+    {
+        String nameWithSuffix = name;
+        if (getFileSuffix() != null && getFileSuffix() != "")
+        {
+          if (name.contains("."))
+          {
+            String[] split = name.split("\\.");
+            String extension = "." + split[split.length-1];
+            nameWithSuffix = name.replace(extension, getFileSuffix() + extension);
```

Comment:
This block is not reliable. What about `tar.gz` or similar?

Reply:
You're right, I'll fix this